### PR TITLE
drop heavy deps

### PR DESCRIPTION
**Pull request by knaaptime**
this PR helps move toward the release:

- drop hard dependencies on pandana, urbanaccess and osmnx
- add project_gdf function inspired by osmnx to utils
- add docstrings to convenience plotting methods
- add citation to segregation profiles
- allow kernel weights in seg_profile to accept different kernel functions

### CODE REVIEW

File: segregation/util/util.py [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
+__author__ = "Levi Wolf <levi.john.wolf@gmail.com>, Renan X. Cortes <renanc@ucr.edu>, and Eli Knaap <ek@knaaptime.com>"
 
 import numpy as np
 import pandas as pd
 import libpysal
 import geopandas as gpd
+import math
 from warnings import warn
-from libpysal.weights import Queen, KNN, Kernel
+from libpysal.weights import Kernel
```
Comment (renanxcortes) [inline segregation/util/util.py]: I think `_return_length_weighted_w` uses KNN, could we import here?

**Comment by renanxcortes** [comment]
Also, @knaaptime, do you know why this is conflicting with util?

File: segregation/util/util.py (hunk 1 above)
Comment (knaaptime) [reply segregation/util/util.py]: we're using the full path from libpysal
```w_aux = libpysal.weights.KNN.from_dataframe(
            data,
            ids=data.index.tolist(),
            geom_col=data._geometry_column_name,
            k=1)
```
so we dont need to import KNN again

**Comment by knaaptime** [comment]
> Also, @knaaptime, do you know why this is conflicting with util?

its a name thing (comma versus "and"). I'll resolve

File: segregation/util/util.py (hunk 1 above)
Comment (renanxcortes) [reply segregation/util/util.py]: It makes sense!

**Comment by knaaptime** [comment]
the remaining errors are 3.7 only:
```
UnsatisfiableError: The following specifications were found to be incompatible with each other:
  - conda-forge/linux-64::boost-cpp==1.70.0=ha2d47e9_0 -> libboost[version='<0']
  - conda-forge/linux-64::libkml==1.3.0=h4fcabce_1010 -> boost-cpp[version='>=1.70.0,<1.70.1.0a0'] -> libboost[version='<0']
  - conda-forge/linux-64::python==3.7.3=h5b0a415_0
  - urbanaccess -> geopy[version='>=1.11.0']
Note that strict channel priority may have removed packages required for satisfiability.
The command "conda install --yes --file requirements_tests.txt;" failed and exited with 1 during .
Your build has been stopped.
```

those look like a handful of incompatible C libraries, and maybe the 3.7 versions havent made it to conda-forge yet, so our channel_priority=strict may be causing a problem? I'm not sure

**Comment by renanxcortes** [comment]
> the remaining errors are 3.7 only:
> 
> ```
> UnsatisfiableError: The following specifications were found to be incompatible with each other:
>   - conda-forge/linux-64::boost-cpp==1.70.0=ha2d47e9_0 -> libboost[version='<0']
>   - conda-forge/linux-64::libkml==1.3.0=h4fcabce_1010 -> boost-cpp[version='>=1.70.0,<1.70.1.0a0'] -> libboost[version='<0']
>   - conda-forge/linux-64::python==3.7.3=h5b0a415_0
>   - urbanaccess -> geopy[version='>=1.11.0']
> Note that strict channel priority may have removed packages required for satisfiability.
> The command "conda install --yes --file requirements_tests.txt;" failed and exited with 1 during .
> Your build has been stopped.
> ```
> 
> those look like a handful of incompatible C libraries, and maybe the 3.7 versions havent made it to conda-forge yet, so our channel_priority=strict may be causing a problem? I'm not sure

hm... if that's the case what would be the best strategy? Could we drop 3.7 from the tests?

**Comment by knaaptime** [comment]
we could pip install that file instead of conda installing it. lemme try that

**Comment by renanxcortes** [comment]
Cool, the pip solution worked! ;)

**Pull request merged**